example_plugins/pair_plugin/pytest/test_example_pair.py

```diff
@@ -9,18 +9,11 @@
 
 import itertools
 import pytest
-from typing import Optional, Tuple
 import numpy as np
-from numpy.typing import NDArray, ArrayLike
 
 
 # Python implementation of the pair force and energy.
-def harm_force_and_energy(
-        dx: ArrayLike,
-        k: float,
-        sigma: float,
-        r_cut: float,
-        shift: Optional[bool] = False) -> Tuple[NDArray[np.float64], float]:
+def harm_force_and_energy(dx, k, sigma, r_cut, shift=False):
```

Comment:
Why this change?

Reply:
The above `from numpy.typing import NDArray, ArrayLike` fails in the Python 3.6 test image: https://github.com/glotzerlab/hoomd-blue/runs/8122103142?check_suite_focus=true#step:6:28

This error was not caught in #1382 because the Python 3.6 tests were not run on that pull request. I changed the build configurations to always include the older python versions in c256cb9 to catch these types of errors sooner.

Reply:
In [ARCHITECTURE.md](https://github.com/glotzerlab/hoomd-blue/blob/trunk-patch/ARCHITECTURE.md#hardware-and-software-support) I state that we support 
> Compilers and software dependencies available on the oldest maintained Ubuntu LTS release. 
That is currently Ubuntu 18.04 which installs Python 3.6 by default.

Following this documentation, we can require Python 3.7 and support Ubuntu 20.04+ starting in April 2023.